src/composables/transactionTx.ts

```diff
@@ -30,33 +27,31 @@ export function useTransactionTx({
   tx,
   externalAddress,
 }: UseTransactionOptions) {
-  const outerTx = ref<ITx | undefined>(tx);
-  const innerTx = ref<ITx | undefined>(tx ? getInnerTransaction(tx) : undefined);
-  const ownerAddress = ref<string | undefined>(externalAddress);
+  const {
+    innerTx,
+    outerTx,
+    setTransactionTx,
+  } = useSplitTransaction({ tx });
 
-  function setTransactionTx(newTx: ITx) {
-    outerTx.value = newTx;
-    innerTx.value = getInnerTransaction(newTx);
-  }
+  const {
+    getOwnershipAccount,
+  } = useOwnership({ tx, store });
 
-  function setExternalAddress(address: string) {
-    ownerAddress.value = address;
-  }
+  const {
+    isDex,
+  } = useDex({ tx, store });
+
+  const {
+    direction,
+  } = useDirectionOfTransaction({ store, tx, externalAddress });
```

Comment:
As mentioned above - I'm not convinced by the idea of using 4 separate composables for one thing that's always used together